edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/KubernetesModule.cs

```diff
@@ -287,9 +289,10 @@ protected override void Load(ContainerBuilder builder)
             builder.Register(
                     async c =>
                     {
-                        var moduleStateStore = c.Resolve<IEntityStore<string, ModuleState>>();
+                        var moduleStateStoreTask = c.Resolve<Task<IEntityStore<string, ModuleState>>>();
```

Comment:
This code isn't as clean as just awaiting on moduleStateStore after resolving IRestartPolicyManager and IRunInfoProvider. I recommend we consolidate to one variable again.

Reply:
Thanks, I changed it on the chance this was part of the problem.  I'll change it back and test it out.